processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregatorFactory.java

```diff
@@ -213,17 +209,16 @@ public AggregatorFactory getMergingFactory(AggregatorFactory other) throws Aggre
   @Override
   public List<AggregatorFactory> getRequiredColumns()
   {
-    return Lists.transform(
-        fields,
-        new Function<DimensionSpec, AggregatorFactory>()
-        {
-          @Override
-          public AggregatorFactory apply(DimensionSpec input)
-          {
-            return new CardinalityAggregatorFactory(input.getOutputName(), Collections.singletonList(input), byRow);
-          }
-        }
-    );
+    return fields.stream()
+                 .map(
+                     input -> new CardinalityAggregatorFactory(
```

Comment:
IMO it could be written denser. Maybe extract "fieldToFactory()" method and condense getRequiredColumns() to one-liner

Reply:
I'd rather leave it like this, since `getRequiredColumns` is already quite short and clear, and a `fieldToFactory` method wouldn't be used anywhere else.

And hopefully this will all be moot eventually. getRequiredColumns is a gross method, only used for groupBy v1 for the flawed "transfer aggregator" concept. I hope we can get rid of it one day.